### ISSUE TITLE
Add option to format whole project at once

### ISSUE THREAD

**Issue by Roald87** [Barteling/STweep#4]
Currently all files have to be manually formatted one by one. Not such a big deal for new projects, but or existing ones that can be quite a lot of work.

Would be nice if there would be an option for format a whole project at once. For safety you could add a check like I did in *TcBlack* with the [`--safe`](https://github.com/Roald87/TcBlack#usage) option, to ensure no unintended changes were made. Although this option is quite slow, because it builds the project before and after formatting. This is done such that it can compare the compile info file name before and after. This filename I think does not change if there are changes which do not influence the execution of the code. For example is you add a comment, add whitelines or change the spacing, the filename doesn't change.

**Comment by Barteling** [comment]
This is a feature I'm working on. In STweep for Codesys this is already in beta. 
There will be an option in the solution explorer context menu to format a node and all of it's child nodes. For example a POU with it's child nodes like methods and properties, or an complete PLC project. 

Performance might be an issue especially to keep the undo function available, this is probably limited to a number of files (for example 20) after that no undo available. you will get an warning though.

Comparing parse trees might be an option, I do this in my (2000+) unit tests (STweep has it's own parser), but actually I believe the formatter will be stable enough. This also might get tricky with your other suggestions like #5 and #1 


Example formatting from the explorer in Codesys:
![1](https://user-images.githubusercontent.com/33071638/104316781-44943000-54dd-11eb-91fd-ac108fef36ec.png)

**Comment by Roald87** [comment]
> This also might get tricky with your other suggestions like #5 and #1

Yeah I guess as long as you only add or remove white space. not much can go wrong. Although we saw that with `VAR_INST` it actually removed an important bracket. 

Good to hear you're already working on it!

**Comment by wla-git** [comment]
I also like this feature for TwinCAT

Is it possible to implement the option to add an attribute {STweep.ignore} to certain declarations and/or implementations? 
sometimes we explicity want to deviate from the standard format

**Comment by Barteling** [comment]
@wilfreddelange , yep working on this. This is also disscused in #14

**Comment by Barteling** [comment]
This is available since 3.6.5
https://www.stweep.com/releases/
